### PR TITLE
chore (types): small refactorings and two new helper types

### PR DESCRIPTION
**Pull request by filmaj**
- adding and removing some TODOs
- factored out some helper types and exported them: `ColorScheme`, `MaxItemsSelectable`, `RichTextBorderable`, `URLRespondable`, and `Conversation`
- inching towards tackling breaking changes / prep for major release of types

### CODE REVIEW

File: packages/types/src/block-kit/blocks.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ export interface Block {
  */
 export type KnownBlock = ImageBlock | ContextBlock | ActionsBlock | DividerBlock |
 SectionBlock | InputBlock | FileBlock | HeaderBlock | VideoBlock | RichTextBlock;
+
```
Comment (zimeg) [inline packages/types/src/block-kit/blocks.ts:53]: 🙌 🙌

File: packages/types/src/block-kit/composition-objects.ts [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
 
+/**
+ * Re-usable labels for common color schemes present in Slack. `danger` displays with a red background (red text on
+ * mobile), while `primary` displays with a green background (green text on mobile).
+ */
+export type ColorScheme = 'primary' | 'danger';
+
+/** The conversation type as available within the Slack UI. */
+export type Conversation = 'im' | 'mpim' | 'private' | 'public';
+
```
Comment (zimeg) [inline packages/types/src/block-kit/composition-objects.ts:11]: Nicee 👏

File: packages/types/src/block-kit/composition-objects.ts [hunk 3]
```diff
@@ -164,15 +173,13 @@ export interface MrkdwnElement {
...
 
-type Conversation = 'im' | 'mpim' | 'private' | 'public';
-
 interface BaseConversationFilter {
   /**
   * @description Indicates which type of conversations should be included in the list. When this field is provided, any
   * conversations that do not match will be excluded. You should provide an array of strings from the following options:
   * `im`, `mpim`, `private`, and `public`. The array cannot be empty.
   */
-  include?: [Conversation, ...Conversation[]]; // TS gymnastics for "at least one item"
```
Comment (zimeg) [inline packages/types/src/block-kit/composition-objects.ts:175]: TIL! This is interesting 😳

File: packages/types/src/message-attachments.ts [hunk 4]
```diff
@@ -182,6 +180,7 @@ export interface Confirmation {
   title?: string;
 }
 
+// Used in web-api chat.* API method request parameters
```
Comment (zimeg) [inline packages/types/src/message-attachments.ts:183]: 👍

**Review by zimeg — APPROVED** [review]
All great changes and great improvements 🪄 The docs are solid too - LGTM! 🚀

File: packages/types/src/block-kit/composition-objects.ts [hunk 5]
```diff
@@ -1,5 +1,14 @@
 // This file contains objects documented here: https://api.slack.com/reference/block-kit/composition-objects
 
+/**
+ * Re-usable labels for common color schemes present in Slack. `danger` displays with a red background (red text on
+ * mobile), while `primary` displays with a green background (green text on mobile).
+ */
+export type ColorScheme = 'primary' | 'danger';
+
+/** The conversation type as available within the Slack UI. */
+export type Conversation = 'im' | 'mpim' | 'private' | 'public';
```
Comment (seratch) [inline packages/types/src/block-kit/composition-objects.ts:10]: How about renaming this to ConversationType or something like that to make it clearer? Once we export it, renaming it later will be a challenge.

**Review by seratch — APPROVED** [review]

**Pull request merged**